torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -158,6 +158,8 @@ c10::optional<std::string> ScriptTypeParser::parseBaseTypeName(
       // Special case for torch.Tensor
       if (isTorch(select.value()) && name == "Tensor") {
         return "Tensor";
+      } else if (isTorch(select.value()) && name == "device") {
+        return "Device";
```

Comment:
Internal fbcode Detectron2 tests break without it because they use `-> torch.device` as a return type annotation.

Reply:
That should work bc of https://github.com/pytorch/pytorch/blob/master/torch/jit/annotations.py#L313. I think what's happening here is the resolver for properties is not being set properly. We should be closing over `torch` bc it's used in the function, but we're not. 

Sometimes with python wrapped functions you have to unpack the wrapped function to do resolution correctly, as in https://github.com/pytorch/pytorch/pull/40468. 

Could you add a test where a property getterr function calls into a free-function ? 
```
def foo(x):
    return x + 1
...

@property
def attr(self):
      return foo(self.a)
```

that should check whether the resolver is being properly set

Reply:
short repro:
```
class Boxes(object):
    @property
    def device(self) -> torch.device:
        return 1 + hi


closure = torch._jit_internal.get_closure(Boxes.device.fget)
# torch is closed over
print(closure)
```

Reply:
I added a test for this but I'm not sure if it right...

Reply:
No, the test looks good. We shouldn't need to add the torch.device thing, it means something else has gone wrong... It would be nice to fix but if resolution in general is working, then I think we can land this. 

Could you remove this "torch.device" logic here, and then in the D2 test change the property to look like ? 
```
from torch import device

@property
def foo() -> device:
    ...
```
This should workaround the problem without added unnecessary stuff to our code base...